pkg/sql/sqlbase/rowfetcher.go

```diff
@@ -266,6 +267,13 @@ func (rf *RowFetcher) NextKey(ctx context.Context) (rowDone bool, err error) {
 				// interleaved data from some other table or index.
 				continue
 			}
+		} else {
+			// We still need to consume the key until the family id, so processKV can
+			// know whether we've finished a row or not.
+			rf.keyRemainingBytes, err = keys.GetFamilySuffix(rf.kv.Key)
+			if err != nil {
+				return false, err
+			}
```

Comment:
Don't quite follow the need for this. I wonder if we need to update a test in sqlbase as well

Reply:
The reason is that `rf.keyRemainingBytes` is used by `processKV` to calculate whether a row has been finished or not. That's required since each row can be split up into multiple column families, each of which lives in its own key.

I would love to augment some rowfetcher tests. However, they do not exist. Adding a unit test suite for rowfetcher would be a lot of work and is probably out of scope for this bug fix. On the plus side, rowfetcher is extensively tested, since it is used by every single sql feature that exercises any table or index reads.

Reply:
Gotcha. Didn't know you introduced mustDecodeIndexKey in the prior change; I was surprised this bug was around all along, and it wasn't. Thanks